src/components/Dashboard/FilesList.tsx

```diff
@@ -60,172 +60,139 @@ export default function FilesList(props: FilesListProps): JSX.Element {
     ref.update({ hidden: !file.hidden });
   };
 
+  const downloadAll = (files: File[]) => {
+    files.forEach(file => {
+      firebase
+        .database()
+        .ref('files/' + file.id)
+        .get()
+        .then((snap: firebase.database.DataSnapshot) => {
+          const fileData = snap.val();
+          console.log('fileData:');
+          console.log(fileData);
+          const code = 'TODO'; // extract from fileData?
```

Comment:
TODO: get code from YJS? idk how YJS works @thecodingwizard

Reply:
for legacy IDE, would want to iterate over `editor-[lang]` for each `lang` and obtain the code for each of them

Reply:
For YJS:

- Edit https://github.com/cpinitiative/ide-yjs/blob/main/src/index.ts to add a route to access / download a file. Specifically, https://github.com/cpinitiative/ide-yjs/blob/764c6277af0e2f6c23e6719590d500eb91ce122f/src/index.ts#L64 here `targetDoc.getText("monaco")` probably will give you the code.
- in this repository, add an api route similar to /copyFile, but instead of copying the file it just returns the contents of a particular file
- in this file, call the api route to get the file

If we wanted to be really fancy, we could even make the YJS server API handler return a ZIP file containing all of that users' files, so we only have to make one API request to get all the files.

Reply:
For Firepad:

https://firepad.io/docs/#headless is probably what we want.